modules/distribution/src/bin/openssl-tls.sh

```diff
@@ -0,0 +1,320 @@
+if [ -z "$JAVA_HOME" ]; then
+    echo "JAVA_HOME is not set. Please set it before running this script."
+    exit 1
+fi
+
+BUILD_OPENSSL=false
+BUILD_PQCLIB=false
+
+# Parse command line arguments
+for arg in "$@"; do
+    if [ "$arg" = "--build_openssl" ] || [ "$arg" = "-build_openssl" ] || [ "$arg" = "build_openssl" ]; then
+        BUILD_OPENSSL=true
+        echo "[Mode]: Building OpenSSL"
+    elif [ "$arg" = "--build_pqclib" ] || [ "$arg" = "-build_pqclib" ] || [ "$arg" = "build_pqclib" ]; then
+        BUILD_PQCLIB=true
+        echo "[Mode]: Building OQS Provider"
+    fi
+done
+
+# Check for build dependencies
+check_dependencies make cmake wget tar gcc
+
+if [ $BUILD_OPENSSL = false ]; then
+    OPENSSL_VERSION=$(openssl version | awk '{print $2}')
+    case "$OPENSSL_VERSION" in
+        3.*)
+            echo "OpenSSL version $OPENSSL_VERSION"
+            ;;
+        *)
+            echo "Requires OpenSSL version 3.0.0 or higher. Found version $OPENSSL_VERSION"
+                    exit 1
+            ;;
+    esac
+    if [ "$(uname)" != "Darwin" ]; then
+        # Check if libssl-dev is installed
+        if [ -f "/etc/debian_version" ]; then
+            if ! dpkg -s libssl-dev > /dev/null 2>&1; then
+                echo "libssl-dev is not installed. Please install it before running this script."
+                exit 1
+            fi
+        else
+            if ! rpm -q openssl-devel > /dev/null 2>&1; then
+                echo "openssl-devel is not installed. Please install it before running this script."
+                exit 1
+            fi
+        fi
+        # Check if libapr1-dev is installed
+        if [ -f "/etc/debian_version" ]; then
+            if ! dpkg -s libapr1-dev > /dev/null 2>&1; then
+                echo "libapr1-dev is not installed. Please install it before running this script."
+                exit 1
+            fi
+        else
+            if ! rpm -q apr-devel > /dev/null 2>&1; then
+                echo "apr-devel is not installed. Please install it before running this script."
+                exit 1
+            fi
+        fi
+    fi
+    check_dependencies apr-1-config
+fi
+
+TMP_DIR="$CARBON_HOME"/openssl-tmp
+if [ -d "$TMP_DIR" ]; then
+    rm -rf "$TMP_DIR"
+fi
+mkdir -p "$TMP_DIR"
+
+# =============================== Install OpenSSL ================================================
+
+if [ $BUILD_OPENSSL = true ]; then
+    OPENSSL_INSTALL_DIR="$CARBON_HOME/native/openssl"
+elif [ "$(uname)" = "Darwin" ]; then
+    if [ "$(uname -m)" = "arm64" ]; then
+        # For M1 (ARM) architecture
+        OPENSSL_INSTALL_DIR=$(ls -d '/opt/homebrew/Cellar/openssl@3/'*)
+    else
+        # For Intel (x86_64) architecture
+        OPENSSL_INSTALL_DIR=$(ls -d '/usr/local/Cellar/openssl@3/'*)
+    fi
+fi
+
+if [ $BUILD_OPENSSL = true ] && [ ! -d "$OPENSSL_INSTALL_DIR" ]; then
+    install_openssl "$TMP_DIR" "$OPENSSL_INSTALL_DIR"
+else
+    echo "OpenSSL is already installed"
+fi
```

Comment:
This block can be refactored into a parent `if [ $BUILD_OPENSSL = true ];` block.